SignalShareExtension/ShareViewController.swift

```diff
@@ -3,26 +3,56 @@
 //
 
 import UIKit
-import Social
 
 import SignalMessaging
 import PureLayout
 import SignalServiceKit
 
-class ShareViewController: SLComposeServiceViewController {
+class ShareViewController: UIViewController {
+
+    override func loadView() {
+        super.loadView()
+
+        // This should be the first thing we do.
+        SetCurrentAppContext(ShareAppExtensionContext())
+
+//        DebugLogger.shared().enableTTYLogging()
```

Comment:
I hope you don't mind but there's some commented-out rough draft code here.  Presumably it will be superceded by your work, but I thought I'd leave it in for now.